woocommerce/woocommerce-gateway-stripe
# Add missing customer ID to subscriptions before processing payment

Fixes #1072.

Address subscriptions that might not have a Stripe customer ID when renewing. This could happen if the customer updated their recurring payment method (see issue #1072). PR #1106 proposed a fix for new recurring payment updates. This one aims to address existing subscriptions that cannot be renewed and have to be manually updated by merchants/customers.

#### Changes proposed in this Pull Request:
- Add missing customer ID to subscriptions before processing payment

#### Testing instructions:
1. Purchase a subscription through a non-Stripe payment method using a version before #1106 
2. Go to My Account > Payment Methods and add a new Stripe payment method
3. Be sure to select the option for "Update the Payment Method used for all of my active subscriptions (optional)."
4. Go to the subscription and click "Billing Details" to edit
5. See that no Customer ID token has been saved to the subscription
6. Attempt to process the renewal for that subscription
    - It should fail since there's no Customer ID
6. After replicating on this branch you should see the renewal was successful and the customer token is set for future ones
-------------------
